admin/resources/school/SchoolCreate.tsx

```diff
@@ -40,6 +48,62 @@ const SchoolCreate = () => {
     redirect(`/school`);
   };
 
+  const onError = async (err: any) => {
+    if (err.toString() == 'HttpError: Uniqueness violation. duplicate key value violates unique constraint "location_pkey"') {
+      let locationRes: any = await clientGQL(`
+        query locationQuery {
+          location(where: {district: {_eq: "${data.district}"}, _and: {block: {_eq: "${data.block}"}, _and: {cluster: {_eq: "${data.cluster}"}}}}) {
+            id
+          }
+        }
+      `)
+      let res = await locationRes.json();
+      if (res?.data?.location?.[0]?.id) {
+        let schoolRes: any = await clientGQL(`
+        mutation insert_school($objects: [school_insert_input!]!) {
+          data: insert_school(objects: $objects) {
+            returning {
+              enroll_count
+              id
+              is_active
+              latitude
+              longitude
+              location_id
+              name
+              session
+              type
+              udise
+              __typename
+            }
+            __typename
+          }
+        }
+        `, {
+          objects: {
+            enroll_count: data.enroll_count ? data.enroll_count : null,
+            is_active: data.is_active ? data.is_active : false,
+            latitude: data.latitude ? data.latitude : null,
+            longitude: data.longitude ? data.longitude : null,
+            location_id: res.data.location[0].id,
+            name: data.name ? data.name : null,
+            session: data.session ? data.session : null,
+            type: data.type ? data.type : null,
+            udise: data.udise ? data.udise : null,
+          }
+        })
+        schoolRes = await schoolRes.json();
+        if (schoolRes?.data?.data?.returning?.[0]) {
+          notify(`School created successfully`, { type: 'success' });
+          redirect(`/school`);
+        } else if (schoolRes.errors) {
+          notify(`${schoolRes.errors[0].message}`, { type: 'error' });
```

Comment:
Always use optional chaining for such condition like schoolRes?.errors?.[0]?.message